lib/tapioca/helpers/env_helper.rb

```diff
@@ -8,10 +8,20 @@ module EnvHelper
 
     requires_ancestor { Thor }
 
+    private
+
     sig { params(options: T::Hash[Symbol, T.untyped]).void }
     def set_environment(options) # rubocop:disable Naming/AccessorMethodName
-      ENV["RAILS_ENV"] = ENV["RACK_ENV"] = options[:environment]
+      set_all_environment_variables(options[:env])
+      ENV["RAILS_ENV"] = ENV["RACK_ENV"] = options[:environment] if options[:environment]
       ENV["RUBY_DEBUG_LAZY"] = "1"
     end
+
+    sig { params(options: T.nilable(T::Hash[String, String])).void }
+    def set_all_environment_variables(options) # rubocop:disable Naming/AccessorMethodName
+      (options || {}).each do |key, value|
+        ENV[key] = value
```

Comment:
So the `--env` values are overriding the `ENV`, is this really what we want?

If so, I think we should document this in the option description or the README.

Reply:
Also, if this is indeed what we want, is there a reason not to use `ENV.merge!(options)`?

Reply:
Yeah, I didn't think of `ENV.merge!`, that sounds better!

Reply:
> So the `--env` values are overriding the `ENV`, is this really what we want?

What else could they be doing? Tapioca sets those env variables as the user has passed them and we don't care if they override some values or not. `bin/tapioca dsl --env FOO:42` is the same thing that happens when someone runs `FOO=42 bin/tapioca dsl`, in either case we override whatever the value of the `FOO` env variable might have been to make it `42` for the duration of that run.